packages/flutter/lib/src/services/raw_keyboard_linux.dart

```diff
@@ -39,32 +39,31 @@ class RawKeyEventDataLinux extends RawKeyEventData {
   /// (GLFW, GTK, QT, etc) may have a different key code mapping.
   final KeyHelper keyHelper;
 
+  /// A string with one or more unicode scalar values in a single keystroke. Two values are handled
+  /// at most. Otherwise, an assertion will be raised. This is typically the character
+  /// that [keyCode] would produce without any modifier keys. For dead keys, it is typically
+  /// the diacritic it would add to a character. Defaults to an empty string, asserted to be not null.
+  final String unicodeScalarValuesProduced;
```

Comment:
This could be a 32-bit int that packs the first two values into it.  It's all we can handle anyhow, and it would make it seem less like you can pack any number of values into it (which is one of Ian's concerns). I realize it would mean changing the engine, but then it definitely wouldn't give the impression that it can be an arbitrary-length string.

Reply:
Changed to an int. Made the relevant changes on the engine repo as well https://github.com/flutter/engine/pull/9386